### ISSUE TITLE
Unicode-objects must be encoded before hashing - update from 1.8 to 1.9.1

### ISSUE THREAD

**Issue by IrinaAlexandra** [jschneier/django-storages#843]
So we've recently updated from version to 1.8 to version 1.9.1. Now we have a `TypeError: Unicode-objects must be encoded before hashing` when saving to `s3`. 

Code: 
```
file = ContentFile("")
document.save_csv_file(file) #writes to csv file using csv library
example_django_model.file.save(file_name, file, save=False)
```

I then proceed to receive the same error as in #708 . Even though that issue was closed a long time ago (with the solution of doing ContentFile(b'')), the code above was working fine in version 1.8. What has changed in 1.9.1 that is now causing this encoding error?

**Comment by LincolnPuzey** [comment]
Did you also update your botocore/boto3 version?

On 2018-11-30 `botocore` published [version 1.12.57](https://github.com/boto/botocore/blob/develop/CHANGELOG.rst#11257) with the change "bugfix:s3: Add md5 header injection to new operations that require it"

I haven't looked at the actual code that changed there but I suspect that botocore is now calculating md5 hashes of uploads.

**Comment by mertcangokgoz** [comment]
any Update?

**Comment by jschneier** [comment]
Fixed by #911
